### PR TITLE
Fix/user datagrid

### PR DESCRIPTION
**Pull request by frenautvh**

### CODE REVIEW

File: client/app/dedicatedCloud/user/dedicatedCloud-user.controller.js [hunk 1]
```diff
@@ -1,132 +1,23 @@
-angular.module("App").controller("DedicatedCloudUserCtrl", function ($scope, $stateParams, $timeout, $q, $translate, DedicatedCloud, Alerter, constants) {
+angular.module("App").controller("DedicatedCloudUserCtrl", function ($scope, $state, $stateParams, $timeout, $translate, $q, DedicatedCloud, ouiDatagridService) {
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/user/dedicatedCloud-user.controller.js]: $q should be the first angular service (if set in alphabtetical order)

File: client/app/dedicatedCloud/user/dedicatedCloud-user.controller.js [hunk 2]
```diff
@@ -135,131 +26,44 @@ angular.module("App").controller("DedicatedCloudUserCtrl", function ($scope, $st
...
-                        namespace: "dedicatedCloud.user.update.poll",
-                        task: operations,
-                        user: infos.user,
-                        successSates: ["canceled", "done"],
-                        errorsSates: ["error"]
-                    });
-                }
+    this.loadUser = ({ id }) => $q.all({
+        user: DedicatedCloud.getUserDetail($stateParams.productId, id)
+    }).then((infos) => {
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/user/dedicatedCloud-user.controller.js]: Destructive pattern could fit no ? 
(If you set `.then({ user })`)

File: client/app/dedicatedCloud/user/dedicatedCloud-user.controller.js [hunk 3]
```diff
@@ -135,131 +26,44 @@ angular.module("App").controller("DedicatedCloudUserCtrl", function ($scope, $st
...
+        infos.user.state = infos.user.state.toUpperCase();
+        infos.user.activationState = infos.user.activationState.toUpperCase();
+        return infos.user;
+    });
 
-                return infos.user;
-            });
+    this.modifyUserRights = (user) => {
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/user/dedicatedCloud-user.controller.js]: Destructive pattern too :)

File: client/app/dedicatedCloud/user/dedicatedCloud-user.controller.js [hunk 4]
```diff
@@ -135,131 +26,44 @@ angular.module("App").controller("DedicatedCloudUserCtrl", function ($scope, $st
...
+        $state.go("app.dedicatedClouds.users.rights", { userId: user.userId });
     };
 
-    $scope.onTransformItemDone = function () {
-        $scope.loading = false;
+    this.editUser = (user) => {
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/user/dedicatedCloud-user.controller.js]: Destructive pattern too :)

File: client/app/dedicatedCloud/user/edit/dedicatedCloud-user-edit.html [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+        <oui-field data-label="{{:: 'dedicatedCloud_USER_token_validator' | translate }}">
+            <label class="oui-switch">
+                <input id="tokenValidator"
+                       name="tokenValidator"
+                       type="checkbox"
+                       class="oui-switch__checkbox"
+                       data-ng-model="$ctrl.user.tokenValidator">
+                <div class="oui-switch__slider"></div>
+                <i class="oui-icon oui-icon-success" aria-hidden="true"></i>
```
Comment (JeremyDec) [inline client/app/dedicatedCloud/user/edit/dedicatedCloud-user-edit.html]: Better use `<span>` no ? (Same for those under)

File: client/app/dedicatedCloud/user/edit/dedicatedCloud-user-edit.controller.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    $onInit () {
+        this.loading = true;
+        return this.DedicatedCloud.getUserDetail(
+            this.productId,
+            this.userId
+        ).then((user) => {
+            this.user = user;
+            this.user.tokenValidator = user.isTokenValidator;
+        }).catch((err) => {
+            this.Alerter.error(`${this.$translate.instant("dedicatedCloud_USER_edit_load_error")} ${_.get(err, "message") || err}`, "dedicatedCloud.user.edit");
```
Comment (marie-j) [inline client/app/dedicatedCloud/user/edit/dedicatedCloud-user-edit.controller.js]: Why not directly use `_.get(err, "message", err)` ?

**Comment by antleblanc** [comment]
@frenautvh there is a conflict :)

**Pull request merged**